### DISCUSSION TITLE
RISC-V Support

### DISCUSSION THREAD

**Discussion by balupton** [denoland/deno#15655] [Ideas]
What are the plans or progress towards riscv architecture support for deno?

VisionFive boards are available via a dev program to significant open-source projects, so that could be worthwhile for deno to make headway on this.

My use case is that Dorothy (which is mostly riscv compatible) uses Deno in a few places, and also has a cross-platform installer script for Deno at: https://github.com/bevry/dorothy/blob/master/commands/setup-util-deno

---

- https://github.com/denoland/deno/issues/18702

**Comment by kitsonk** [comment]
The biggest challenge with any architecture is not specifically the availability of the hardware but the challenge of integrating that hardware into the CI pipeline which is currently GitHub Actions based.

In addition a lot of core dependencies would need be ported over to RISC-V to make it work.  It does appear for V8 the [changes have been merged](https://groups.google.com/a/groups.riscv.org/g/sw-dev/c/jQi2yIq7jaA) but there can be a lot of pitfalls and the like with the other tooling to make it smooth.  I believe for other architectures we have accepted patches into [rust_v8](https://github.com/denoland/rusty_v8), but there are still [~540 dependencies](https://deno.land/benchmarks#cargo-deps) that would need be cleanly built against a RISC-V target.

Several of the core team are huge fans of the RISC-V architecture. That being said, would you rather want the core team working on porting Deno to an interesting but still currently hobbyist architecture, or fixing bugs and delivering features that would benefit 99% of the users?  For now, it feels like it would need to be a community effort to get something that builds on RISC-V and likely, if there are changes that need to be merged upstream, they would likely be accepted as PRs.

**Comment by beingminimal** [comment]
@kitsonk is there any progress in this direction?
